### DISCUSSION TITLE
Installation of 5.0+ in Kubernetes cluster?

### DISCUSSION THREAD

**Discussion by ankushg** [tsightler/ring-mqtt#362] [Support Q&A]
Hi all!

now that 5.0 is out, it looks like you can’t provide the ring authentication token via environment variable anymore

does anyone have any tips on how to set up the app in a kubernetes cluster? Maybe there’s a way to use ingress to access the same UI that the home-assistant addon uses?

it’d be extra cool if we could get a semi-official Helm chart — maybe via the k8s-at-home base chart template!

**Comment by tsightler** [comment]
As a persistent volume is now required you can always just pre-stage the ring-state.json with the token on the persistent volume along with the desired config.json file, but, realistically, as much as I love K8S, I wouldn't expect any Helm charts any time soon as, at this time, running in K8S is simply not a goal of this project, but I'll never say never.  That being said, if there is no config.json or ring-state.json file in the container a default config.json will be created and it will start the web UI if no token is found, so you should be able to simply use the kubectl port-forward option to temporarily forward a local port to the app, something like (untested):
```
kubectl port-forward  <your_pod_name> 55123:55123
```

**Reply by ankushg** [reply]
Awesome, that should be enough to at least get me started… 

Does the web UI exist only if there’s no state file, or can it be used to re-authenticate if the state file somehow no longer works?

any thoughts on a web UI for editing config.json?

**Reply by tsightler** [reply]
Yes, web UI only runs when there is no state file with the only exception to this is the addon.  The primary reasoning behind this is that the web UI represents a security risk, it's not authenticated or encrypted and I'm certainly no web developer, so it has limited validation and zero security.  There's also very limited reason to keep it running in the non-addon case since you can just edit the config files directly on the host.  For the addon, I do leave it running as there's slightly less risk since at least it is not directly exposed as it's proxied through the HA frontend which can provide authentication and encryption.

I have considered enhancing the web UI to edit the config.json, but again, it's just not needed for 99.9% of users, the vast majority of users for this project use the HA addon, which already provides a web UI for configuration.  For users leveraging docker or manual install, editing the file directly is just not a major issue.  If the web UI did allow editing the config file then it would need to run all the time and offer some type of authentication and encryption, which gets complex really quickly and I just don't see the benefit.

However, perhaps there is another way.  I have been considering the possibility to allow config.json options to be updated via MQTT.  Perhaps the Docker image could accept MQTT_URL environment variable and then all other configuration options, including the token, could be updated via MQTT.  This way, users could at least choose a secure path via MQTT authentication and encryption, if they wanted.

BTW, I think you could get the web UI to run all the time by just using options.json instead of config.json for the configuration file name.  That's the only indicator that the startup scripts use to determine if the image is running as the HA addon or not.  You might get some errors because the startup script will attempt to detect some parameters from the home assistant UI via bashio, which will of course fail, but these should not be fatal.
